project-base/storefront/urql/cache/updates.ts

```diff
@@ -0,0 +1,134 @@
+            manuallyUpdateCartQuery(cache, newCart, args.input.cartUuid);
+        },
+        RemoveFromCart(result, args: RemoveFromCartMutationVariablesApi, cache) {
+            const newCart =
+                typeof result.RemoveFromCart !== 'undefined' ? (result.RemoveFromCart as CartApi) : undefined;
+            manuallyUpdateCartQuery(cache, newCart, args.input.cartUuid);
+        },
+        ApplyPromoCodeToCart(result, args: ApplyPromoCodeToCartMutationVariablesApi, cache) {
+            const newCart =
+                typeof result.ApplyPromoCodeToCart !== 'undefined'
+                    ? (result.ApplyPromoCodeToCart as CartApi)
+                    : undefined;
+            manuallyUpdateCartQuery(cache, newCart, args.input.cartUuid);
+        },
+        RemovePromoCodeFromCart(result, args: RemovePromoCodeFromCartMutationVariablesApi, cache) {
+            const newCart =
+                typeof result.RemovePromoCodeFromCart !== 'undefined'
+                    ? (result.RemovePromoCodeFromCart as CartApi)
+                    : undefined;
+            manuallyUpdateCartQuery(cache, newCart, args.input.cartUuid);
+        },
+        AddProductToList(result, args: { input: ProductListUpdateInputApi }, cache) {
+            manuallyUpdateProductListQuery(args.input.productListInput, result.AddProductToList, cache);
+        },
+        RemoveProductFromList(result, args: { input: ProductListUpdateInputApi }, cache) {
+            if (result.RemoveProductFromList === null) {
+                manuallyRemoveProductListQuery(cache, args.input.productListInput);
+            } else {
+                manuallyUpdateProductListQuery(args.input.productListInput, result.AddProductToList, cache);
+            }
+        },
+        CleanProductList(_result, args: CleanProductListMutationVariablesApi, cache) {
+            manuallyRemoveProductListQuery(cache, args.input);
+        },
+    },
+};
+
+const manuallyUpdateCartQuery = (cache: Cache, newCart: CartApi | undefined, cartUuid: string | null) => {
+    if (newCart) {
+        cache.updateQuery({ query: CartQueryDocumentApi, variables: { cartUuid } }, (data) => {
+            const updatedData = data || { __typename: 'Cart', cart: null };
+            updatedData.cart = newCart;
+
+            return updatedData;
+        });
+    }
+};
+
+const manuallyRemoveProductListQuery = (cache: Cache, args: ProductListInputApi) => {
+    const query =
+        args.type === ProductListTypeEnumApi.WishlistApi ? WishlistQueryDocumentApi : ComparisonQueryDocumentApi;
+
+    cache.updateQuery({ query: query, variables: { input: args } }, (data) => ({
+        ...data,
+        __typename: 'ProductList',
+        productList: null,
+    }));
+};
+
+const manuallyUpdateProductListQuery = (input: ProductListInputApi, result: DataField, cache: Cache) => {
+    const query =
+        input.type === ProductListTypeEnumApi.WishlistApi ? WishlistQueryDocumentApi : ComparisonQueryDocumentApi;
```

Comment:
Nope. This is not good approach in order of this refactoring. We don't want to refactor this after adding third or next type. Let's refactor this so it is prepared for next list.

Reply:
This was completely removed 👍